src/Orleans.Core/Messaging/Message.cs

```diff
@@ -9,6 +10,20 @@
 
 namespace Orleans.Runtime
 {
+    internal static class EventSourceUtils
+    {
+        public static void SetEventSourceActivityIdIfNotNull(this Message msg)
+        {
+            if(msg.ActivityId != null)
+                EventSource.SetCurrentThreadActivityId(msg.ActivityId.Guid);
```

Comment:
Unsure, but do we need to track the existing thread activity id so we can restore it on completion?

Also: formatting